lib/Lift/InstructionTranslator.cpp

```diff
@@ -826,60 +744,80 @@ IT::translateOpcode(PTCOpcode Opcode,
 
   using v = std::vector<Value *>;
   switch (Opcode) {
-  case PTC_INSTRUCTION_op_movi_i32:
-  case PTC_INSTRUCTION_op_movi_i64:
-    return v{ ConstantInt::get(RegisterType, ConstArguments[0]) };
-  case PTC_INSTRUCTION_op_discard:
+  case LIBTCG_op_discard:
     // Let's overwrite the discarded temporary with a 0
     return v{ ConstantInt::get(RegisterType, 0) };
-  case PTC_INSTRUCTION_op_mov_i32:
-  case PTC_INSTRUCTION_op_mov_i64:
-    return v{ Builder.CreateTrunc(InArguments[0], RegisterType) };
-  case PTC_INSTRUCTION_op_setcond_i32:
-  case PTC_INSTRUCTION_op_setcond_i64: {
+  case LIBTCG_op_mov_i32:
+  case LIBTCG_op_mov_i64:
+    if (auto *Constant = dyn_cast<ConstantInt>(InArguments[0])) {
+      return v{ Constant };
+    } else {
+      return v{ Builder.CreateTrunc(InArguments[0], RegisterType) };
+    }
+  case LIBTCG_op_setcond_i32:
+  case LIBTCG_op_setcond_i64: {
+    revng_assert(ConstArguments.size() > 0 and
+                 ConstArguments[0].kind == LIBTCG_ARG_COND);
     Value *Compare = createICmp(Builder,
-                                ConstArguments[0],
+                                ConstArguments[0].cond,
                                 InArguments[0],
                                 InArguments[1]);
     // TODO: convert single-bit registers to i1
+    // NOTE(anjo) I don't think this is a good idea, semantics of QEMU require
+    // these to be full size, this is especially true for negsetcond
```

Comment:
Is this comment about the TODO above?

Reply:
Yes exactly, QEMU has no concept of i1, so to match semantics we should just use i32 or i64